cloudigrade/analyzer/management/commands/analyze_log.py

```diff
@@ -136,3 +170,36 @@ def _is_on_off_event(self, record, valid_events):
             return False
         else:
             return True
+
+    @transaction.atomic
+    def _save_instances_and_events(self, instances):
+        """
+        Save instances and on/off events to the DB.
+
+        Args:
+            instances (dict): Instance data, including a list of events.
+
+        """
+        for instance_id, instance_dict in instances.items():
+            account_id = instance_dict['account_id']
+            # TODO: Don't pull the same account record from the
+            #       DB multiple times. Consider caching.
+            account = Account.objects.get(account_id=account_id)
```

Comment:
I realize its a todo, but is there a reason you didn't pull lines 184 and 187 outside the for loop? Doesn't look like they have to be in the loop.

Reply:
Only because I'm a baddy. 🙁